blockToAddresses.go

```diff
@@ -213,58 +227,103 @@ func getTraceAddresses(addresses map[string]bool, traces *BlockTraces, blockNum
+                //if isGood(author) {
+                //  addresses[author + "\t" + blockNum + "\t" + "99998"] = true
+                //}
             } else {
                 fmt.Println("New type of reward", traces.Result[i].Action.RewardType)
             }
         } else if traces.Result[i].Type == "suicide" {
             // add the contract that died, and where it sent it's money
             address := traces.Result[i].Action.Address
+            if isGood(address) {
+                addresses[address + blockAndIdx] = true
+            }
             refundAddress := traces.Result[i].Action.RefundAddress
-            addresses[address + blockAndIdx] = true
-            addresses[refundAddress + blockAndIdx] = true
+            if isGood(refundAddress) {
+                addresses[refundAddress + blockAndIdx] = true
+            }
+
         } else if traces.Result[i].Type == "create" {
             // add the creator, and the new address name
             from := traces.Result[i].Action.From
+            if isGood(from) {
+                addresses[from + blockAndIdx] = true
+            }
             address := traces.Result[i].Result.Address
-            addresses[from + blockAndIdx] = true
-            addresses[address + blockAndIdx] = true
+            if isGood(address) {
+                addresses[address + blockAndIdx] = true
+            }
 
-            // If it's a top level trace, then the call data is the init, 
+            // If it's a top level trace, then the call data is the init,
             // so to match with quickblocks, we just parse init
             if len(traces.Result[i].TraceAddress) == 0 {
                 if len(traces.Result[i].Action.Init) > 10 {
                     initData := traces.Result[i].Action.Init[10:]
                     for i := 0; i < len(initData) / 64; i++ {
                         addr := string(initData[i * 64:(i + 1) * 64])
                         if isPotentialAddress(addr) {
-                            addresses["0x" + string(addr[24:]) + blockAndIdx] = true
+                            addr = "0x" + string(addr[24:])
+                            if isGood(addr) {
+                                addresses[addr + blockAndIdx] = true
+                            }
                         }
                     }
                 }
             }
 
             // How can we check if the contract creation has failed?
             // If the contract throws during construction, then I don't get that address
-            // If this has failed, then I can get the 
-
+            // If this has failed, then I can get the
+
+            // Handle contract creations that error out
+            if traces.Result[i].Action.To == "" {
+                if traces.Result[i].Result.Address == "" {
+                    if traces.Result[i].Error != "" {
+                        bytes, err := getTransactionReceipt(traces.Result[i].TransactionHash)
+                        if err != nil {
```

Comment:
Ideally, we'd set up a data flow where this can be it's own Go routine. It makes a call to the RPC, and so is probably just hanging out for a while. 

Does this fix the difference issue?

Reply:
E.g. does this correctly handle the case where contract address should be included?